### PR TITLE
rancher2 cluster fixes

### PR DESCRIPTION
**Pull request by rawmind0**
This PR contains:
* Fix `rancher2_cluster` resource, fixing provider crash if `cluster_monitoring_input` argument is deleted
* Improved `rancher2_cluster_sync` with new cluster state check method and new option to wait until alerting is enabled

Issues https://github.com/rancher/terraform-provider-rancher2/issues/627 and https://github.com/rancher/terraform-provider-rancher2/issues/633

### CODE REVIEW

File: rancher2/config.go [hunk 1]
```diff
@@ -1111,16 +1139,57 @@ func (c *Config) UpdateClusterByID(cluster *managementClient.Cluster, update map
...
+				return true, obj, nil
+			}
+			return false, obj, nil
+		}
+	}
+
+	return false, obj, nil
+}
+
+func (c *Config) isClusterMonitoringEnabled(id string) (bool, *managementClient.Cluster, error) {
```
Comment (superseb) [inline rancher2/config.go]: Not really sure about this naming, this check if cluster monitoring is Active right?

File: rancher2/config.go [hunk 2]
```diff
@@ -1111,16 +1139,57 @@ func (c *Config) UpdateClusterByID(cluster *managementClient.Cluster, update map
...
+				return true, obj, nil
+			}
+			return false, obj, nil
+		}
+	}
+
+	return false, obj, nil
+}
+
+func (c *Config) isClusterAlertingEnabled(id string) (bool, *managementClient.Cluster, error) {
```
Comment (superseb) [inline rancher2/config.go]: Same as with `isClusterMonitoringEnabled`

Comment (rawmind0) [reply rancher2/config.go]: This function is checking if cluster condition, `clusterAlertingEnabledCondition   = "AlertingEnabled"` is true

File: rancher2/config.go (hunk 1 above)
Comment (rawmind0) [reply rancher2/config.go]: yeah, this is a question how cluster state is checked. This function is checking if cluster condition, `clusterMonitoringEnabledCondition = "MonitoringEnabled"` is true

Comment (superseb) [reply rancher2/config.go]: Can we change the function name to include condition so its clear it is checking the condition being enabled and not the config cluster monitoring? Or does that not make sense?

Comment (rawmind0) [reply rancher2/config.go]: Updated function name to `isClusterMonitoringEnabledCondition`

File: rancher2/config.go (hunk 2 above)
Comment (rawmind0) [reply rancher2/config.go]: Updated function name to `isClusterAlertingEnabledCondition`

**Review by superseb — APPROVED** [review]

**Pull request merged**